nemo/collections/nlp/models/dialogue/dialogue_gpt_classification_model.py

```diff
@@ -612,8 +677,11 @@ def prepare_data(self):
 
     def setup(self, stage=None):
         super().setup()
-        if self.cfg.library == "megatron" and self.prompt_learning:
-            self.language_model.init_new_prompts()
+        if self.cfg.library == "megatron" and self.prompt_learning and stage == "fit":
+            if self.cfg.virtual_prompt_style == 'prompt-tuning':
```

Comment:
Can you use the enum check here instead of comparing strings? Like how we do here: https://github.com/NVIDIA/NeMo/blob/42c7a671dea3ef767ddeaf6891861ff830106c42/nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py#L179

Same comment for the if statement in line 292

Reply:
Thanks for catching this